src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DebugDirectoryNode.cs

```diff
@@ -114,6 +129,21 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                 builder.EmitReloc(entry, RelocType.IMAGE_REL_FILE_ABSOLUTE);
             }
 
+            // Second emit a record for the perfmap
```

Comment:
Nit - at the first glance I would be inclined to put this code in <code>PerfMapDebugDirectoryEntryNode</code> as it deals with internals of the node like its length that's also declared there. But I see that it's not done that way for <code>NativeDebugDirectoryEntryNode</code> so I don't insist.

Reply:
I can do that, I'll do it as an add-on cleanup commit that I can ignore when porting this to the release branch.